pkg/config/example_test.go

```diff
@@ -0,0 +1,265 @@
+/*
+Copyright IBM Corp All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package config_test
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"log"
+	"math/big"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	cb "github.com/hyperledger/fabric-protos-go/common"
+	ob "github.com/hyperledger/fabric-protos-go/orderer"
+	pb "github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/hyperledger/fabric/pkg/config"
+)
+
+// fetchChannelConfig mocks retrieving the most recent configuration block
+func fetchChannelConfig() *cb.Config {
+	// How to retrieve block using the cli: https://hyperledger-fabric.readthedocs.io/en/release-1.4/commands/peerchannel.html#peer-channel-fetch-example.
+	// How to convert to JSON: https://hyperledger-fabric.readthedocs.io/en/release-1.4/commands/configtxlator.html#configtxlator-proto-decode.
+	return &cb.Config{
+		ChannelGroup: &cb.ConfigGroup{
+			Groups: map[string]*cb.ConfigGroup{
+				config.OrdererGroupKey: {
+					Version: 1,
+					Groups: map[string]*cb.ConfigGroup{
+						"OrdererOrg": {
+							Groups: map[string]*cb.ConfigGroup{},
+							Values: map[string]*cb.ConfigValue{
+								config.KafkaBrokersKey: {
+									ModPolicy: config.AdminsPolicyKey,
+									Value: marshalOrPanic(&ob.KafkaBrokers{
+										Brokers: []string{"kafka0:9092", "kafka1:9092"},
+									}),
+								},
+							},
+							Policies: map[string]*cb.ConfigPolicy{
+								config.AdminsPolicyKey: {
+									ModPolicy: config.AdminsPolicyKey,
+									Policy: &cb.Policy{
+										Type: 3,
+										Value: marshalOrPanic(&cb.ImplicitMetaPolicy{
+											Rule:      cb.ImplicitMetaPolicy_MAJORITY,
+											SubPolicy: config.AdminsPolicyKey,
+										}),
+									},
+								},
+							},
+							ModPolicy: config.AdminsPolicyKey,
+						},
+					},
+					Values: map[string]*cb.ConfigValue{
+						config.BatchTimeoutKey: {
+							Value: marshalOrPanic(&ob.BatchTimeout{
+								Timeout: "15s",
+							}),
+						},
+						config.BatchSizeKey: {
+							Value: marshalOrPanic(&ob.BatchSize{
+								MaxMessageCount:   100,
+								AbsoluteMaxBytes:  100,
+								PreferredMaxBytes: 100,
+							}),
+						},
+					},
+					Policies: make(map[string]*cb.ConfigPolicy),
+				},
+				config.ApplicationGroupKey: {
+					Groups: map[string]*cb.ConfigGroup{
+						"Org1MSP": {
```

Comment:
should this be org name "Org1" instead of MSPID?

Reply:
Yes, it'd be more accurate.

Reply:
🛠